### ISSUE TITLE
Cannot read property 'name' of undefined in "MaterialTopTabView"

### ISSUE THREAD

**Issue by sagar-tomar-groww** [react-navigation/react-navigation#11891]
### Current behavior

We are getting intermittent crash on Bugsnag citing the above errorMessage. When I am trying to reproduce the error, I am not able to right now but it's quite a significant chunk of the total crashes being reported for a specific version. Seems like intermittent issue though, the only catch i noticed about the screens where it's happening is that we are trying to navigate to that screen with "PUSH" as navigation type

**STACKTRACE**

TypeError Cannot read property 'name' of undefined 
    node_modules/@react-navigation/material-top-tabs/src/views/MaterialTopTabView.tsx:52:31 onIndexChange
    node_modules/react-native-tab-view/src/TabView.tsx:66:6 onIndexChange
    node_modules/react-native-tab-view/src/PagerViewAdapter.tsx:165:10 onPageSelected
    node_modules/react-native-pager-view/src/PagerView.tsx:87:17 anonymous
    (native) apply
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:22:9 invokeGuardedCallbackImpl
    (native) apply
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:40:28 invokeGuardedCallback
    (native) apply
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:53:24 invokeGuardedCallbackAndCatchFirstError
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:73:2 executeDispatch
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1125:8 executeDispatchesAndReleaseTopLevel
    (native) call
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:361:58 forEachAccumulated
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1160:6 anonymous
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:8457:11 batchedUpdatesImpl
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1106:11 batchedUpdates
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1137:2 _receiveRootNodeIDEvent
    node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1178:4 receiveEvent
    (native) apply
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:428:26 value
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:113:11 anonymous
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:368:8 value
    node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:112:9 value
    
    --------------------------------------------------------------------------------------------
    
    
    
<img width="1294" alt="Screenshot 2024-03-13 at 3 27 31 PM" src="https://github.com/react-navigation/react-navigation/assets/113590267/e0e280d1-0bb2-4d0d-a211-174179a544d2">

### Expected behavior

There should no crash as such. 

### Reproduction

Not able to repro

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [X] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 6.1.6
| @react-navigation/bottom-tabs          |  6.0.5
| @react-navigation/material-top-tabs    | 6.0.2
| @react-navigation/native-stack         | 6.6.2
| react-native-safe-area-context         | 4.5.3
| react-native-screens                   | 3.21.0
| react-native-gesture-handler           | 2.9.0
| react-native-reanimated                | 3.5.4
| react-native-tab-view                  | 3.5.2
| react-native-pager-view                | 5.4.21
| react-native                           | 0.72.5
| node                                   | 18.15.0
| npm or yarn                            | 1.22.18

**Comment by sagar-tomar-groww** [comment]
I have already tried 6.6.12 but since the issue is not properly reproducible but rather intermittent so can't really push 6.6.12 just like that in production. 

Checked https://github.com/react-navigation/react-navigation/pull/10721 but as @satya164 correctly figured out that it'll suppress the root cause of the issue. 
@satya164 can someone from the team help us in debugging this and remove this crash as it's impacting our stability numbers.
